### DISCUSSION TITLE
Latest version features

### DISCUSSION THREAD

**Discussion by rsssxd** [bromite/bromite#1983] [Q&A]
- How exactly do the following two features work?
- How does the Linux feature work?

• restrict creation of dynamic code
• forbid dynamic code generation on Linux

**Comment by qua3k** [comment]
The design and implementation is laid out very clearly in https://docs.hexavalent.org/sandboxing/dynamic-code.html. I’ve updated our patch on the development branch to make it not reliant on our own V8 feature flag but that hasn’t been merged into our main branch yet. I’m not sure why he merged these patches as-is serious discussion with the original authors and without knowledge of our upcoming improvements that were intended to make it more browser-agnostic.

**Comment by csagan5** [comment]
> * How does the Linux feature work?

~They are not Linux-specific.~ Edit: I thought so after [this comment](https://github.com/GrapheneOS/Vanadium/issues/170#issuecomment-1099657092).

> I’m not sure why he merged these patches as-is serious discussion with the original authors

What do you mean by serious discussion?

**Reply by qua3k** [reply]
> They are not Linux-specific.

They are specific to Linux desktop. The idea could be applied to Android as well by finding some way to hook the baseline policy as Android doesn't currently have fine-grained seccomp filters like on desktop.

> What do you mean by serious discussion?

There should have been more discussion with the authors about how the patches design/function before deciding to merge them. I could have made it more clear that the idea was applicable to Android, not the patches as-is.

**Reply by csagan5** [reply]
> > What do you mean by serious discussion?
> 
> There should have been more discussion with the authors about how the patches design/function before deciding to merge them. I could have made it more clear that the idea was applicable to Android, not the patches as-is.

I prefer to embed as much documentation/links as possible in patch descriptions so that interpretation does not need to go through anyone; in this case I was misled by the short conversation in https://github.com/GrapheneOS/Vanadium/issues/170#issuecomment-1099657092, I also initially thought they were not applicable to Android.

Thanks for developing these patches/ideas, I will continue participating in the discussion there (or elsewhere appropriate).

**Comment by rsssxd** [comment]
Can you explain how the Linux feature works for Android?
@csagan5

**Reply by csagan5** [reply]
@rsssxd that patch does nothing on Android, the changed files are not even compiled; I was misled by the discussion we had on another issue and then I thought it was complementary to the other two.
